### PR TITLE
DP-7628 & DP-7706: Upgrade SelectBox Atom and OrgSelector Molecule to new standards

### PR DESCRIPTION
**Pull request by avrilpearl**
Upgrade the SelectBox component's story to meet new standards.

1. component's story include proper knob labels, all stored in props passed to component
2. adds readme file for component
3. adds comments for component props, should show on the component info pane

### CODE REVIEW

File: src/components/atoms/forms/SelectBox/SelectBox.stories.js [hunk 1]
```diff
@@ -2,29 +2,25 @@ import React from 'react';
...
-      const onChangeCallback = () => {
-        console.log('This is a custom onChange callback passed to the select!');
+    withInfo(`<div>${SelectBoxDocs}</div>`)(() => {
+      const defaultHandler = () => { console.log('This is a custom onChange callback passed to the select!'); };
+      const props = {
+        label: text('selectBox.label', 'Color Scheme:'),
+        required: boolean('selectBox.required', true),
+        id: text('selectBox.id', 'color-select'),
+        options: object('selectBox.options', selectOptions.options.colors),
+        onChangeCallback: button('selectBox.onChangeCallback', defaultHandler)
```
Comment (mrjmd) [inline src/components/atoms/forms/SelectBox/SelectBox.stories.js]: I see the console message fire when I click the button down in the knobs section of Storybook, but I'm not seeing it when I make a drop-down selection. I should be, right?

**Comment by mrjmd** [comment]
screen shot 2018-02-06 at 1 29 12 pm

**Comment by avrilpearl** [comment]
@mrjmd --> so this is some odd knobs add on behavior, i converts nums to string, and then throw an error when you pass an array of props to a subcomponent. It still renders on the front-end - so not sure how to handle resolving this error.

File: src/components/atoms/forms/SelectBox/SelectBox.stories.js (hunk 1 above)
Comment (avrilpearl) [reply src/components/atoms/forms/SelectBox/SelectBox.stories.js]: @mrjmd, i guess adding the button know make the onchangecallback function not be called how it was set up in the component - i updated the branch so this is triggered on change - but the button is no longer available in the knobs section.

File: src/components/atoms/forms/SelectBox/SelectBox.stories.js [hunk 2]
```diff
@@ -2,29 +2,25 @@ import React from 'react';
...
-      // selectOptions imported above
-      const options = object('options', selectOptions.options.colors);
-
-      const defaultValue = true;
-      const required = boolean('required', defaultValue);
-
-      const onChangeCallback = () => {
-        console.log('This is a custom onChange callback passed to the select!');
+    withInfo(`<div>${SelectBoxDocs}</div>`)(() => {
+      const defaultHandler = () => { console.log('This is a custom onChange callback passed to the select!'); };
```
Comment (smurrayatwork) [inline src/components/atoms/forms/SelectBox/SelectBox.stories.js]: Shouldn't this get handled by the action logger?

Comment (smurrayatwork) [reply src/components/atoms/forms/SelectBox/SelectBox.stories.js]: defaultHandler is also not used anywhere in this story.

Comment (avrilpearl) [reply src/components/atoms/forms/SelectBox/SelectBox.stories.js]: @smurrayatwork --> i have remove this from the story, and per your suggestion, added an action logger event.

**Pull request merged**